plugins/ci/skills/diagnose-job-run-symptoms/diagnose_job_run.py

```
@@ -0,0 +1,218 @@
+        if "log in" in body.lower():
+            return None, ("got an SSO login page instead of JSON — token is "
+                          "missing/expired; use the oc-auth skill to refresh it")
+        return None, "gateway returned an HTML error page (likely 504 timeout); retry later"
+    try:
+        return json.loads(body), None
+    except ValueError:
+        return None, "server returned a non-JSON response body"
+
+
+def get_json(url):
+    with urllib.request.urlopen(url, timeout=30) as resp:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)